### DISCUSSION TITLE
Drag file onto string field in the inspector

### DISCUSSION THREAD

**Discussion by Dukembg** [godotengine/godot-proposals#8016] [Editor]
I'm pretty noob here, and searched all over to see if anyone has suggested this.

Since you can drag files into GD script as a way to copy its path, I feel like this should also be true when dragging a file over a string field in the inspector. Also, I noticed that Godot has a super awesome UID string for each file that can really save a dev when major changes happen in a project and was wondering why not just use the UID by default?

https://github.com/godotengine/godot-proposals/assets/98967543/14175158-7823-48a7-a569-bcc6ed6de648

**Comment by Zireael07** [comment]
You can already do that with a NodePath field.

**Reply by Dukembg** [reply]
It doesn't seem to work with the files, only what's in the tree hierarchy.

**Comment by aXu-AP** [comment]
This is supported, you need to use `@export_file` annotation. It also has functionality to filter files and adds file open dialog next to the text field. Check docs: https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_exports.html#strings-as-paths

So for your example the correct way would be:

    @export_file("*.tscn") var any_scene : String

**Reply by Dukembg** [reply]
Perfect!
